### PR TITLE
Code scaffolding

### PR DESCRIPTION
**Pull request by evan10s**
Adds backend/frontend templates:

Backend:
- node (v16+ recommended)
- yarn
- Express
- Typescript

Frontend:
- node (v16+ required)
- yarn
- Vue3
- Typescript

Both can be run locally using `yarn run dev` in the respective directory, and there's basic linting (+ prettier for vue because it offered it) as well

### CODE REVIEW

File: server/src/constants/HttpStatusCodes.ts [hunk 1]
```diff
@@ -0,0 +1,386 @@
...
+     * The server cannot meet the requirements of the Expect request-header field.
+     */
+    EXPECTATION_FAILED = 417,
+
+    /**
+     * This code was defined in 1998 as one of the traditional IETF April Fools' jokes, in RFC 2324, Hyper Text Coffee Pot Control Protocol,
+     * and is not expected to be implemented by actual HTTP servers. The RFC specifies this code should be returned by
+     * teapots requested to brew coffee. This HTTP status is used as an Easter egg in some websites, including Google.com.
+     */
+    I_AM_A_TEAPOT = 418,
```
Comment (khari05) [inline server/src/constants/HttpStatusCodes.ts:274]: How applicable is HTTP Teapot in this use case?

~~Completely ignore that I randomly stumbled upon this repo then started reading a PR~~

Comment (evan10s) [reply server/src/constants/HttpStatusCodes.ts:274]: 418 is a great status code, but this is actually just some generated code from running [express-generator-typescript](https://www.npmjs.com/package/express-generator-typescript)

Comment (khari05) [reply server/src/constants/HttpStatusCodes.ts:274]: lol

File: server/.eslintrc.json [hunk 2]
```diff
@@ -0,0 +1,44 @@
+{
+  "parser": "@typescript-eslint/parser",
+  "plugins": [
+    "@typescript-eslint"
+  ],
+  "extends": [
+    "eslint:recommended",
+    "plugin:@typescript-eslint/recommended",
+    "plugin:@typescript-eslint/recommended-requiring-type-checking",
+    "plugin:node/recommended"
```
Comment (khari05) [inline server/.eslintrc.json:10]: Maybe just use an all-encompassing eslint ruleset like ts-standard?

File: server/src/repos/database.json [hunk 3]
```diff
@@ -0,0 +1 @@
+{"users":[{"name":"Sean Maxwell","email":"sean.maxwell@gmail.com","pwdHash":"$2b$12$1mE2OI9hMS/rgH9Mi0s85OM2V5gzm7aF3gJIWH1y0S1MqVBueyjsy","role":1,"id":159123164363},{"name":"Gordan Freeman","email":"gordan.freeman@halflife.com","pwdHash":"$2b$12$1mE2OI9hMS/rgH9Mi0s85OM2V5gzm7aF3gJIWH1y0S1MqVBueyjsy","role":0,"id":906524522143},{"name":"John Smith","email":"jsmith@yahoo.com","pwdHash":"$2b$12$1mE2OI9hMS/rgH9Mi0s85OM2V5gzm7aF3gJIWH1y0S1MqVBueyjsy","role":0,"id":357437875835}]}
```
Comment (khari05) [inline server/src/repos/database.json:1]: should this be in source control?

File: server/src/views/login.html [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+        Password:<br/>
+        <input id="pwd-input" type="password"/>
+      </div>
+
+      <div>
+        <button id="login-btn">Login</button>
+      </div>
+
+      <div class="default-login-user">
+        // Default admin user (Email: "sean.maxwell@gmail.com", Password: "Password@1")
```
Comment (khari05) [inline server/src/views/login.html:28]: fyi this is here

File: server/src/repos/database.json (hunk 3 above)
Comment (evan10s) [reply server/src/repos/database.json]: this is nothing more than dummy code from express-generator-typescript and will be removed (along with much other the other base code in a future PR). I pulled out most of the auth code here

**Pull request merged**